packages/ethereum-contracts/contracts/ECDSA_Verify.sol

```diff
@@ -0,0 +1,112 @@
+//SPDX-License-Identifier: Unlicense
```

Comment:
where's this adapted from - did you write part/all of the comment below?

if we're just using the adapted version to avoid installing the OZ package, we might be better off just importing the package

Reply:
Yes I wrote the comment below.

OK I figured out a much cleaner implementation that imports more of the OZ code instead of just inlining it.

Reply:
Great!